### ISSUE TITLE
AppBar and ToolBar: Overflow button is not present if there are no commands in the overflowarea.

### ISSUE THREAD

**Issue by AmazingJaze** [winjs/winjs#982]
There is a bug in the AppBar/ToolBar where the overflowbutton hides itself if the overflowarea is empty. This code is based off the old toolbar control that only ever had 1 display mode for the action area that allowed you to see all commands and their labels.

Unless the AppBar/ToolBar has both closedDisplayMode: 'full', and no commands in the overflowarea, the overflowbutton should be visible and clickable.

This is most severe with closedDisplayMode: "minimal" since users can see the AppBar but are unable to see or interact with their primary commands.

**Comment by lamuertepeluda** [comment]
If there is no command with section: 'secondary' the appBar (in minimal status) won't open.

**Comment by lamuertepeluda** [comment]
I don't know if this is connected, but I have an app in closed status minimal, with 2 buttons, one of which is hidden.
If I don't put any button with section = 'secondary' (which I don't need), then hide it, the app will behave correctly only until the first primary command click.
After clicking on one of the primary commands, the three dots disappears.

**Comment by lamuertepeluda** [comment]
Furthermore if the app is closed in minimal mode, the buttons get the 'inline-block' display which shows them even though they should be not visible.

I put this in my stylesheet even if it is just a patch

    .win-commandingsurface-closeddisplayminimal.win-commandingsurface-overflowtop.win-commandingsurface-closed .win-command {
      display: none !important;
    }

**Comment by AmazingJaze** [comment]
Hi @lamuertepeluda 
I am actively working on this issue right now. The first issue you raise is absolutely related, but can you help me better understand the second issue? 

Are you saying that with the following markup you can see the AppBarCommand even while the AppBar is closed?

```HTML
<div data-win-control="WinJS.UI.AppBar data-win-options="{closedDisplayMode: 'minimal'}">
   <button data-win-control="WinJS.UI.AppBarCommand" data-win-options={icon: 'add', label: 'add'}>
</div>
```

**Comment by AmazingJaze** [comment]
edited previous reply for syntax.

**Comment by lamuertepeluda** [comment]
Hi AmazingJaze,

yes what happens is that I see icons being displayed at half, but I'm using custom svg icons like this

<button data-win-control="WinJS.UI.AppBarCommand"
           data-win-res="{winControl: {label: 'login'}}"
           data-win-options="{id:'loginCmd', icon:'url(images/appbar/login.svg)' , type:'button', section:'primary', label:'Login', extraClass: 'customicon'}"></button>

**Comment by AmazingJaze** [comment]
Hi @lamuertepeluda. I am surprised you would see that behavior. Are you sure you're not using custom CSS that is overriding the default AppBar styles for closedDisplayMode minimal?

https://github.com/winjs/winjs/blob/26bee7a9c606f362dcb7e4189b04ac99663a16e8/src/less/styles-commandingsurface.less#L94-L100

All .win-commands in the action area should be 
``` display: none; ``` not ``` display: inline-block; ```

If you use those same icons in an AppBar on a separate test page with a clean version of WinJS, do you still see that behavior? I'd be curious to know why the display: none styles are getting overridden

**Comment by lamuertepeluda** [comment]
no, I remember this 'inline-block' being set on the style property of the
element, probably from within the appbar code.

sadly I'm not at the PC (holiday).

this didn't occur with winjs 4.0.1
Il 12/ago/2015 09:36 AM, "AmazingJaze" <notifications@github.com> ha
scritto:

**Comment by AmazingJaze** [comment]
Hi @lamuertepeluda  I am unable to repro locally the 2nd issue locally and this fix for #982 has been checked in and this thread has been closed. 

If you find that you are still able to repro  your second issue and that the inline-block styles are not coming from custom CSS in your app, please file a separate bug, and feel free to reference this conversation.

**Comment by AmazingJaze** [comment]
updated for clarity.
